src/gwt/src/org/rstudio/studio/client/workbench/views/environment/dataimport/DataImport.java

```diff
@@ -63,4 +104,27 @@ public DataImportOptions getOptions()
       
       return options;
    }
+   
+   private void refreshPreview()
+   {
+      server_.previewDataImport(fileChooserTextBox_.getTextBoxText(), maxRows_, new ServerRequestCallback<JsObject>()
```

Comment:
K! @jmcphers recommended to introduce a delay as well since for most local operations don't take long enough for progress to show completely. Therefore, added a ProgressIndicatorDelay class which I'm using from this class.